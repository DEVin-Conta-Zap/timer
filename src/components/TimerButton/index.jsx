import './styles.css';
import PropTypes from 'prop-types';

const TimerButton = ({ buttonAction, buttonValue}) => {

  return (
    <div className='button-container' onClick={() => buttonAction()}>
      <p>{buttonValue}</p>
    </div>
  )
}

TimerButton.propTypes = {
  buttonAction: PropTypes.func.isRequired,
  buttonValue: PropTypes.string.isRequired
};

export default TimerButton;