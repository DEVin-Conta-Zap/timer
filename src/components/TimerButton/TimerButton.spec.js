import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event';
import TimerButton from '.'

describe('<TimerButton />', () => {
  it('should renders a button', () => {
    const {container} = render(<TimerButton buttonValue="Save" buttonAction={jest.fn()}/>);
    
    expect(screen.getByText('Save')).toBeInTheDocument();
    // eslint-disable-next-line testing-library/no-node-access
    expect(container.firstChild).toMatchSnapshot();
  })

  it('should call a method when button is clicked', () => {
    const saveAction = jest.fn();

    render(<TimerButton buttonAction={saveAction} buttonValue="Save"/>);

    userEvent.click(screen.getByText('Save'));

    expect(saveAction).toBeCalledTimes(1);
  })
})