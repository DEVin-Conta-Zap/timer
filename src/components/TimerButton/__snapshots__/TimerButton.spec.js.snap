// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimerButton /> should renders a button 1`] = `
<div
  class="button-container"
>
  <p>
    Save
  </p>
</div>
`;
